libs/designsystem/src/lib/components/modal/services/modal.helper.spec.ts

```diff
@@ -114,6 +147,133 @@ describe('ModalHelper', () => {
+            size: 'small',
+          });
+          ionModal = await ionModalController.getTop();
+          expect(ionModal).toBeTruthy();
+          expect(ionModal.classList.contains('small')).toBeTrue();
+          expect(ionModal.classList.contains('medium')).toBeFalse();
+          expect(ionModal.classList.contains('large')).toBeFalse();
+          await overlay.dismiss();
+        });
+
+        it('modal should be sized `medium`', async () => {
+          overlay = await modalHelper.showModalWindow({
+            title: 'Modal',
+            component: InputEmbeddedComponent,
+            size: 'medium',
+          });
+          ionModal = await ionModalController.getTop();
+          expect(ionModal).toBeTruthy();
+          expect(ionModal.classList.contains('small')).toBeFalse();
+          expect(ionModal.classList.contains('medium')).toBeTrue();
+          expect(ionModal.classList.contains('large')).toBeFalse();
+          await overlay.dismiss();
+        });
+
+        it('modal should be sized `large`', async () => {
+          overlay = await modalHelper.showModalWindow({
+            title: 'Modal',
+            component: InputEmbeddedComponent,
+            size: 'large',
+          });
+          ionModal = await ionModalController.getTop();
+          expect(ionModal).toBeTruthy();
+          expect(ionModal.classList.contains('small')).toBeFalse();
+          expect(ionModal.classList.contains('medium')).toBeFalse();
+          expect(ionModal.classList.contains('large')).toBeTrue();
+          await overlay.dismiss();
+        });
+
+        it('should not set sizing class (large) if flavor is `drawer`', async () => {
+          overlay = await modalHelper.showModalWindow({
+            title: 'Modal',
+            flavor: 'drawer',
+            component: InputEmbeddedComponent,
+          });
+          ionModal = await ionModalController.getTop();
+          expect(ionModal).toBeTruthy();
+          expect(ionModal.classList.contains('small')).toBeFalse();
+          expect(ionModal.classList.contains('medium')).toBeFalse();
+          expect(ionModal.classList.contains('large')).toBeFalse();
+          await overlay.dismiss();
+        });
+
+        it("should add class `content-overflows`, if content can't fit in viewport", async () => {
+          overlay = await modalHelper.showModalWindow({
+            title: 'Modal',
+            component: ContentOverflowsWithFooterEmbeddedComponent,
+          });
+          ionModal = await ionModalController.getTop();
+          const ionModalWrapper = ionModal.querySelector('.modal-wrapper');
+          expect(ionModal).toBeTruthy();
+          expect(ionModalWrapper).toBeTruthy();
+
+          await TestHelper.whenTrue(() => {
+            return ionModalWrapper.classList.contains('content-overflows');
+          });
+
+          expect(ionModalWrapper.classList.contains('content-overflows')).toBeTrue();
+          await overlay.dismiss();
```

Comment:
Consider moving this to `afterEach` to ensure modal is dismissed if `expect` throws an error